packages/atlas/src/hooks/useNftState.ts

```diff
@@ -0,0 +1,65 @@
+import { useMemo } from 'react'
+
+import { useNft } from '@/api/hooks'
+import { useUser } from '@/providers/user'
+
+export const useNftState = (id?: string) => {
+  const { nft, loading } = useNft(id || '')
+  const { memberships } = useUser()
+
+  const nftFetched = nft && !loading
+  const isOwner = !memberships?.find((membership) => membership.handle == nft?.ownerMember?.handle)
+
+  const isBuyNow = useMemo(() => {
```

Comment:
What's the point of all the `useMemo` here? We don't do any complicated operations and we have stable references anyway since all of those are booleans